src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.Fast.cs

```diff
@@ -39,6 +39,20 @@ public static partial class MemoryMarshal
         /// </summary>
         public static ref T GetReference<T>(ReadOnlySpan<T> span) => ref span._pointer.Value;
 
+        /// <summary>
+        /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to fake non-null pointer. Such a reference can be used
+        /// for pinning but must never be dereferenced. This is useful for interop with methods that do not accept null pointers for zero-sized buffers.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe ref T GetNonNullPinnableReference<T>(Span<T> span) => ref (span.Length != 0) ? ref span._pointer.Value : ref Unsafe.AsRef<T>((void*)1);
+
+        /// <summary>
+        /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to fake non-null pointer. Such a reference can be used
```

Comment:
nit: GetReference specifies ReadOnlySpan in the comments, while this does not. Update one or the other.

Reply:
Also, "If the Span is empty, returns a reference to a fake non-null pointer." And what would GetReference do in that case that is different? Can we specify what happens if the span is empty in the comments of GetReference?

Reply:
Fixed.